crates/subspace-networking/src/request_handlers/generic_request_handler.rs

```diff
@@ -86,7 +91,7 @@ impl<Request: GenericRequest> RequestHandler for GenericRequestHandler<Request>
                 pending_response,
             } = request;
 
-            match self.handle_request(peer, payload) {
+            match self.handle_request(peer, payload).await {
```

Comment:
We're not trying to call blocking tasks from async context here, but rather making request handler async, so we can call async stuff from it without `block_on` (see other changes in this PR).

Reply:
OK got it, the sync/async bridging is always messy :-(  Do you see any use case for having multiple requests in flight in the request/response handler, now that these are async/futures (using FuturesUnordered, etc)?

Reply:
It is not just about multiple requests happening in parallel even though it does happen, this will block other libp2p protocols as well since we have some parts of networking which are global in libp2p (events handling) and some that are per-connection (accessing storage provider for instance). Can't recall of top of my head where this one lands, but in either case blocking corresponding executor thread is very undesirable.